src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ProcessLinkerXmlBase.cs

```diff
@@ -38,33 +44,33 @@ public abstract class ProcessLinkerXmlBase
         const string AllAssembliesFullName = "*";
         protected const string XmlNamespace = "";
 
-        protected readonly string _xmlDocumentLocation;
         readonly XPathNavigator _document;
-        protected readonly (EmbeddedResource Resource, AssemblyDefinition Assembly)? _resource;
-        protected readonly LinkContext _context;
+        protected readonly (ManifestResource Resource, ModuleDesc Module)? _resource;
```

Comment:
I think it will be enough to keep just the ModuleDesc here. The resource itself is only used to mark it for removal. We do that in ManifestResourceBlockingPolicy.cs that does all the filtering necessary.